src/extensions/surveys.tsx

```diff
@@ -356,16 +363,20 @@ export function Questions({
         originalQuestionIndex: number
         displayQuestionIndex: number
     }) => {
-        const isLastDisplayedQuestion = displayQuestionIndex === survey.questions.length - 1
+        if (!posthog) {
+            return
+        }
 
         const responseKey =
             originalQuestionIndex === 0 ? `$survey_response` : `$survey_response_${originalQuestionIndex}`
 
-        if (isLastDisplayedQuestion) {
-            return sendSurveyEvent({ ...questionsResponses, [responseKey]: res }, survey, posthog)
+        setQuestionsResponses({ ...questionsResponses, [responseKey]: res })
+
+        const nextStep = posthog?.getNextSurveyStep(survey, displayQuestionIndex, res)
```

Comment:
Here, `nextStep` should always exist right, so 
```suggestion
        const nextStep = posthog.getNextSurveyStep(survey, displayQuestionIndex, res)
```

I'd expect the above to work^ if it doesn't there's potentially a bug? (or ts being dumb)